hack/update-stack-version.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+
+#
+# Update the Elastic Stack version
+# Usage: ./hack/update-stack-version.sh <previous_version> <new_version>
+#
+
+set -eu
+
+# Elastic Stack versions
+PREVIOUS_VERSION="$1"
+NEW_VERSION="$2"
+
+# Use the "-i.bak" trick to be compatible with both Linux and OSX
+bump_version() {
+	sed -i.bak -E "s|${PREVIOUS_VERSION}|${NEW_VERSION}|g" "$1"
```

Comment:
One minor issue with this is that `update-stack-version.sh 7.11.2 7.12.0` will also replace `791192` (or any similar string) with `7.12.0` as well. Not sure whether it's worth worrying about though.

It'd be nice to have a usage message when the script is invoked without any arguments. Maybe that can show how to escape the version to avoid the above problem. (`update-stack-version.sh 7\.11\.2 7.12.0`)

Reply:
Good point. I will add a usage message.

I don't want to make the script to complicated by adding validation. Especially in bash, it quickly becomes a hassle to validate `\.` .

But now I'm thinking that we could derive the previous version to escape the dots. It doesn't sound so bad.

```sh
previous_version() {
	grep STACK_VERSION Makefile | grep -Eo "[0-9]+\.[0-9]+\.[0-9]+" | sed "s|\.|\\\.|g"
}
```

Reply:
Back from WE, I wonder: why not rewrite the script in Go?

Reply:
I think the shell utilities are highly optimized for this problem and anything we write ourselves will require a lot of work to deal with edge cases and performance (evidenced by "I re-implemented X in Y" articles).  So it's probably not worth the effort? The chances of there being a string like `791192` in the code is pretty unlikely, and if it does happen, hopefully it would be very obvious from a quick glance through the diff.

Reply:
Yes, let's stay there, it's already an improvement.